runtime/shared_common/CacheMap.cpp

```diff
@@ -7400,3 +7406,50 @@ SH_CacheMap::updateLocalHintsData(J9VMThread* currentThread, J9SharedLocalStartu
 	}
 	memcpy(&localHints->hintsData, &updatedHintsData, sizeof(J9SharedStartupHintsDataDescriptor));
 }
+
+static void
+checkROMClassUTF8SRPs(J9ROMClass *romClass)
+{
+	if ((UnitTest::CORRUPT_CACHE_TEST == UnitTest::unitTest)
+		|| (UnitTest::CACHE_FULL_TEST == UnitTest::unitTest)
+		|| (UnitTest::PROTECTA_SHARED_CACHE_DATA_TEST == UnitTest::unitTest)
+		|| (UnitTest::PROTECT_NEW_ROMCLASS_DATA_TEST == UnitTest::unitTest)
+	) {
+		return;
+	}
+
+	UDATA romClassEnd = (UDATA)romClass + (UDATA)romClass->romSize;
+	U_32 i = 0;
+
+	Trc_SHR_Assert_True((UDATA)J9ROMCLASS_CLASSNAME(romClass) < romClassEnd);
+	Trc_SHR_Assert_True((UDATA)J9ROMCLASS_SUPERCLASSNAME(romClass) < romClassEnd);
+	Trc_SHR_Assert_True((UDATA)J9ROMCLASS_OUTERCLASSNAME(romClass) < romClassEnd);
+
+	if (romClass->interfaceCount > 0) {
+		J9SRP * interfaceNames = J9ROMCLASS_INTERFACES(romClass);
+		for (i = 0; i < romClass->interfaceCount; i++) {
+			Trc_SHR_Assert_True(NNSRP_PTR_GET(interfaceNames, UDATA) < romClassEnd);
+			interfaceNames++;
+		}
+	}
+	if (romClass->innerClassCount > 0) {
+		J9SRP* innerClassNames = J9ROMCLASS_INNERCLASSES(romClass);
+		for (i = 0; i < romClass->innerClassCount; i++) {
+			Trc_SHR_Assert_True(NNSRP_PTR_GET(innerClassNames, UDATA) < romClassEnd);
+			innerClassNames++;
+		}
+	}
+
+#if defined(J9VM_OPT_VALHALLA_NESTMATES)
+	Trc_SHR_Assert_True((UDATA)J9ROMCLASS_NESTHOSTNAME(romClass) < romClassEnd);
+
+	if (romClass->nestMemberCount > 0) {
+		J9SRP *nestMemberNames = J9ROMCLASS_NESTMEMBERS(romClass);
+		for (i = 0; i < (U_32)nestMemberCount; i++) {
```

Comment:
nestMemberCount looks like an undeclared variable.

Reply:
Fixed. It is inside #ifdef so compiler didn't capture this.